src/main/java/org/jabref/gui/UpdateTimestampListener.java

```diff
@@ -0,0 +1,25 @@
+package org.jabref.gui;
+
+import org.jabref.model.entry.event.EntryChangedEvent;
+import org.jabref.preferences.JabRefPreferences;
+
+import com.google.common.eventbus.Subscribe;
+
+/**
+ * Updates the timestamp of changed entries if the feature is enabled
+ */
+class UpdateTimestampListener {
+    private final JabRefPreferences jabRefPreferences;
+
+    UpdateTimestampListener(JabRefPreferences jabRefPreferences) {
```

Comment:
When you here directly pass an object of  TimestampPreferences, you then don't need to  mock the whole JabRef Preferences and you will get rid of the travis architecture failing

Reply:
Thank you for your suggestion. I already considered the mentioned approach but I could not get it to work.

The problem is described in third paragraph of the section _"Problem with dependency on JabRefPreferences"_ in the initial pull request. To summarize:
When I tested this approach, the passed TimestampPreferences object did not get updated when the settings were changed. This would mean that
- either the application would have to be restarted for the settings change to apply to the listener
- or I would have to add a method to change the held TimestamPreferences object and add a method that listens for changes in the timestamp settings (which might also just move the problems to the tests for the new listener).

Both approaches are not valid in my opinion and I could not think of a better one.

Noticing this problem also lead to my other question concerning the usage of the JabRefPreferences object directly instead of the object held in "Globals.prefs".
(The question was: Is it possible that the held object changes, thereby invalidating the previously held object which would still be used by the UpdateTimestampListener?)

Or should the TimestampPreferences object get updated and it not reflecting the settings changes is not the wanted behavior?

Reply:
I investigated your problem and I know the reason for it.
The registerListener method in BasePanel is called once per Database at the initialization with the current Preferences. Any changes to the prefs will not update this.